### PR TITLE
gnrc_gomach: add a brief description of GoMacH's features.

### PR DESCRIPTION
**Pull request by zhuoshuguo**
### Contribution description

This PR adds a brief description of GoMacH's main features.

### Issues/PRs references
 See also #5618 .

### CODE REVIEW

File: sys/include/net/gnrc/gomach/gomach.h [hunk 1]
```diff
@@ -7,10 +7,25 @@
  */
 
 /**
- * @defgroup    net_gnrc_gomach A traffic-adaptive multi-channel MAC
+ * @defgroup    net_gnrc_gomach GoMacH
  * @ingroup     net_gnrc
- * @brief       A traffic adaptive MAC protocol that provides high traffic
- *              adaptability, high energy efficiency and high robustness.
+ * @brief       GoMacH is, we believe, "a General, nearly Optimal MAC protocol
```
Comment (miri64) [inline sys/include/net/gnrc/gomach/gomach.h]: Do we believie it or know it. Anyway this `, we believe,` should be removed IMHO.

File: sys/include/net/gnrc/gomach/gomach.h [hunk 2]
```diff
@@ -7,10 +7,25 @@
...
+ *              for multi-Hop communications", for IEEE 802.15.4 IoT/WSNs networks.
```
Comment (miri64) [inline sys/include/net/gnrc/gomach/gomach.h]: Multi-hop

File: sys/include/net/gnrc/gomach/gomach.h [hunk 3]
```diff
@@ -7,10 +7,25 @@
...
+ *              It is designed to be a traffic adaptive MAC protocol that provides
+ *              high traffic adaptability, high energy efficiency and high robustness.
+ *
+ * ## GoMacH's main features
+ * 1. it is an asynchronous solution which doesn't rely on global synchronization;
+ * 2. it supports multi-hop and mesh network;
+ * 3. it adopts duty-cycle scheme to conserve power;
+ * 4. it provides high traffic adaptation for handling burst or dynamic traffics.
+ *    It achieves this by dynamically allocating transmission slots for intensive
+ *    senders (that have pending packets); it enables a (or several) sender
```
Comment (miri64) [inline sys/include/net/gnrc/gomach/gomach.h]: one (or more) sender(s)

File: sys/include/net/gnrc/gomach/gomach.h [hunk 4]
```diff
@@ -7,10 +7,25 @@
...
+ *    to burst transmit all of its buffered packets (for the same destination) to
+ *    the receiver in one shot (or during a shot notice); The burst slotted-transmissions
+ *    (of multi senders) will be ordered by the receiver device in a TDMA period;
```
Comment (miri64) [inline sys/include/net/gnrc/gomach/gomach.h]: multiple senders

File: sys/include/net/gnrc/gomach/gomach.h [hunk 5]
```diff
@@ -7,10 +7,25 @@
...
+ * 5. it adopts multi-channel scheme for avoiding/reducing wireless interference jam.
```
Comment (miri64) [inline sys/include/net/gnrc/gomach/gomach.h]: article missing.

File: sys/include/net/gnrc/gomach/gomach.h [hunk 6]
```diff
@@ -7,10 +7,25 @@
...
  * @ingroup     net_gnrc
- * @brief       A traffic adaptive MAC protocol that provides high traffic
- *              adaptability, high energy efficiency and high robustness.
+ * @brief       GoMacH is, we believe, "a General, nearly Optimal MAC protocol
+ *              for multi-Hop communications", for IEEE 802.15.4 IoT/WSNs networks.
+ *              It is designed to be a traffic adaptive MAC protocol that provides
+ *              high traffic adaptability, high energy efficiency and high robustness.
+ *
+ * ## GoMacH's main features
+ * 1. it is an asynchronous solution which doesn't rely on global synchronization;
```
Comment (miri64) [inline sys/include/net/gnrc/gomach/gomach.h]: I think this whole list can be more paraphrased,

```
1. doesn't rely on global synchronization
2. support for multi-hop and mesh networks
....
```

Comment (miri64) [inline sys/include/net/gnrc/gomach/gomach.h]: Also, why it is an enumerated list? These features do not appear to be in any particular order or priority.

**Comment by zhuoshuguo** [comment]
Addressed comments.

File: sys/include/net/gnrc/gomach/gomach.h [hunk 7]
```diff
@@ -7,10 +7,20 @@
...
- * @brief       A traffic adaptive MAC protocol that provides high traffic
- *              adaptability, high energy efficiency and high robustness.
+ * @brief       GoMacH is, "a General, nearly Optimal MAC protocol
+ *              for multi-Hop communications", for IEEE 802.15.4 IoT/WSNs networks.
+ *              It is designed to be a traffic adaptive MAC protocol that provides
+ *              high traffic adaptability, high energy efficiency and high robustness.
+ *
+ * ## GoMacH's main features
+ * doesn't rely on global synchronization;
+ * supports for Multi-hop and mesh network;
```
Comment (miri64) [inline sys/include/net/gnrc/gomach/gomach.h]: Why is Multi-hop capitalized now?

File: sys/include/net/gnrc/gomach/gomach.h [hunk 8]
```diff
@@ -7,10 +7,20 @@
...
  * @ingroup     net_gnrc
- * @brief       A traffic adaptive MAC protocol that provides high traffic
- *              adaptability, high energy efficiency and high robustness.
+ * @brief       GoMacH is, "a General, nearly Optimal MAC protocol
+ *              for multi-Hop communications", for IEEE 802.15.4 IoT/WSNs networks.
+ *              It is designed to be a traffic adaptive MAC protocol that provides
+ *              high traffic adaptability, high energy efficiency and high robustness.
+ *
+ * ## GoMacH's main features
+ * doesn't rely on global synchronization;
```
Comment (miri64) [inline sys/include/net/gnrc/gomach/gomach.h]: Consider removing semi-colons and periods or decide for one of them.

File: sys/include/net/gnrc/gomach/gomach.h (hunk 7 above)
Comment (zhuoshuguo) [reply sys/include/net/gnrc/gomach/gomach.h]: Ah, my bad, mistakenly did it. fix soon

**Comment by zhuoshuguo** [comment]
Addressed comments again~

File: sys/include/net/gnrc/gomach/gomach.h [hunk 9]
```diff
@@ -7,10 +7,20 @@
...
  * @ingroup     net_gnrc
- * @brief       A traffic adaptive MAC protocol that provides high traffic
- *              adaptability, high energy efficiency and high robustness.
+ * @brief       GoMacH is, "a General, nearly Optimal MAC protocol
+ *              for multi-Hop communications", for IEEE 802.15.4 IoT/WSNs networks.
+ *              It is designed to be a traffic adaptive MAC protocol that provides
+ *              high traffic adaptability, high energy efficiency and high robustness.
+ *
+ * ## GoMacH's main features
+ * doesn't rely on global synchronization
```
Comment (miri64) [inline sys/include/net/gnrc/gomach/gomach.h]: mh... now it isn't a list anymore I just realized (the stars are just from the C-comment). Probably most readable to do it with dashes:

```
 *  - doesn't rely on global synchronization
 *  - ...

**Comment by zhuoshuguo** [comment]
Addressed comments again~

**Comment by zhuoshuguo** [comment]
Will squash after ACK~

File: sys/include/net/gnrc/gomach/gomach.h [hunk 10]
```diff
@@ -7,10 +7,25 @@
  */
 
 /**
- * @defgroup    net_gnrc_gomach A traffic-adaptive multi-channel MAC
+ * @defgroup    net_gnrc_gomach GoMacH
  * @ingroup     net_gnrc
- * @brief       A traffic adaptive MAC protocol that provides high traffic
- *              adaptability, high energy efficiency and high robustness.
+ * @brief       GoMacH is, "a General, nearly Optimal MAC protocol
```
Comment (smlng) [inline sys/include/net/gnrc/gomach/gomach.h]: sorry, but this is not _brief_, maybe the original defgroup description (`A traffic-adaptive multi-channel MAC`) can be (re) used here. Please note that the `@brief` sentence will be displayed in doxygen below the group name (GoMacH, here), hence it should be a one-liner. You can use this longer text as as the first paragraph of the detailed description (above the feature list), though.

Comment (smlng) [reply sys/include/net/gnrc/gomach/gomach.h]: to make my point a bit clearer, look at the module [page](https://doc.riot-os.org/modules.html) in doxygen, the `@brief` for the boards group is for instance `Board specific definitions and implementations`.

Comment (miri64) [reply sys/include/net/gnrc/gomach/gomach.h]: @zhuoshuguo you can also look at the compiled doc locally by building with `make doc` in the RIOT root directory. The results will be in `doc/doxygen/html/`. Choose the `module.html` to look at the new version of the page @smlng mentioned.

Comment (miri64) [reply sys/include/net/gnrc/gomach/gomach.h]: (saw this comment only after ACKing)

File: sys/include/net/gnrc/gomach/gomach.h [hunk 11]
```diff
@@ -7,10 +7,25 @@
...
+ *              for multi-Hop communications", for IEEE 802.15.4 IoT/WSNs networks.
+ *              It is designed to be a traffic adaptive MAC protocol that provides
+ *              high traffic adaptability, high energy efficiency and high robustness.
```
Comment (zhuoshuguo) [inline sys/include/net/gnrc/gomach/gomach.h]: @smlng But we have the removed description above texts displayed here~ :-)

Comment (zhuoshuguo) [reply sys/include/net/gnrc/gomach/gomach.h]: Ah, you mean the description should be a `one-liner`, i.e., not longer than one line (like 2 or 3 lies)? @smlng @miri64

Comment (smlng) [reply sys/include/net/gnrc/gomach/gomach.h]: exactly, the text is fine, just add a one-liner for `@brief` and move these 2 sentences after that (2 empty lines) so its part of the longer description.

Comment (zhuoshuguo) [reply sys/include/net/gnrc/gomach/gomach.h]: Thanks!~ Learned and worked on it~ :-)

**Comment by zhuoshuguo** [comment]
Addressed comments and squashed!~

**Comment by smlng** [comment]
compiled docs, looks good!

**Pull request merged**